pymc/tests/test_mixture.py

```diff
@@ -657,6 +658,103 @@ def test_iterable_single_component_warning(self):
         with pytest.warns(UserWarning, match="Single component will be treated as a mixture"):
             Mixture.dist(w=[0.5, 0.5], comp_dists=[Normal.dist(size=2)])
 
+    @pytest.mark.parametrize(
+        "weights, comp_dists, size, expected",
+        [
+            (
+                np.array([1, 0]),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                None,
+                -2,
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                None,
+                2.8,
+            ),
+            (
+                np.array([0.5, 0.5]),
+                [Normal.dist(-2, 5), Exponential.dist(lam=1 / 3)],
+                None,
+                0.5,
+            ),
+            (
+                np.broadcast_to(np.array([0.4, 0.6]), (5, 3, 2)),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                None,
+                np.full(shape=(5, 3), fill_value=2.8),
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5, size=(5, 3)), Normal.dist(6, 3, size=(5, 3))],
+                None,
+                np.full(shape=(5, 3), fill_value=2.8),
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                (3,),
+                np.full(shape=(3,), fill_value=2.8),
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                (11, 7),
+                np.full(shape=(11, 7), fill_value=2.8),
+            ),
+            (
+                # only one to fail, maybe due to https://github.com/ricardoV94/pymc/blob/mixtures/pymc/distributions/mixture.py#L212
+                # I'm not sure...
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5, size=(11, 7)), Normal.dist(6, 3, size=(11, 7))],
+                (5, 3),
+                np.full(
+                    shape=(
+                        11,
+                        7,
+                        5,
+                        3,
+                    ),
+                    fill_value=2.8,
+                ),
+            ),
+            (
+                np.array([0.4, 0.6]),
+                Normal.dist(mu=np.array([-2, 6]), sigma=np.array([5, 3])),
+                None,
+                2.8,
+            ),
+            (
+                np.tile(1 / 13, 13),
+                Normal.dist(-2, 1, size=(13,)),
+                None,
+                -2,
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [
+                    MvNormal.dist(mu=np.array([-1, -2]), cov=np.eye(2) * 0.3),
+                    MvNormal.dist(mu=np.array([3, 5]), cov=np.eye(2) * 0.8),
+                ],
+                None,  # support not yet available for batched multivariate distributions
```

Comment:
What does the comment mean, that it is not yet expected to work?

Reply:
Oh I was implying that, because components are multivariate, I was expecting that this wouldn't work. Just trying to check if I understood